xxd: tslcert1.crt: No such file or directory

Followed the instructions on OpenSuse 15.2, getting the following build error:

```
$ make 
Scanning dependencies of target docs
[  1%] Generating API documentation with Doxygen
warning: ignoring unsupported tag `JAVADOC_BANNER         =' at line 200, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `PYTHON_DOCSTRING       =' at line 228, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `OPTIMIZE_OUTPUT_SLICE  =' at line 300, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `NUM_PROC_THREADS       =' at line 466, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `EXTRACT_PRIV_VIRTUAL   =' at line 492, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `RESOLVE_UNNAMED_PARAMS =' at line 536, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `SHOW_HEADERFILE        =' at line 609, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `WARN_IF_INCOMPLETE_DOC =' at line 827, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `DOCSET_FEEDURL         =' at line 1360, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `FULL_SIDEBAR           =' at line 1571, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `OBFUSCATE_EMAILS       =' at line 1602, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `HTML_FORMULA_FORMAT    =' at line 1613, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `FORMULA_MACROFILE      =' at line 1639, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `MATHJAX_VERSION        =' at line 1661, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `LATEX_MAKEINDEX_CMD    =' at line 1850, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `LATEX_EMOJI_DIRECTORY  =' at line 1985, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `XML_NS_MEMB_FILE_SCOPE =' at line 2119, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `DOT_UML_DETAILS        =' at line 2432, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `DOT_WRAP_THRESHOLD     =' at line 2441, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: ignoring unsupported tag `DIR_GRAPH_MAX_DEPTH    =' at line 2514, file /home/avatar/synology/build/libdigidocpp/build/Doxyfile
warning: argument `TEXT' for option CLASS_GRAPH is not a valid boolean value
Using the default: YES!
[  1%] Built target docs
Scanning dependencies of target minizip
[  2%] Building C object src/CMakeFiles/minizip.dir/minizip/zip.c.o
[  3%] Building C object src/CMakeFiles/minizip.dir/minizip/unzip.c.o
[  4%] Building C object src/CMakeFiles/minizip.dir/minizip/ioapi.c.o
[  5%] Linking C static library libminizip.a
[  5%] Built target minizip
Scanning dependencies of target digidocpp_util
[  7%] Building CXX object src/CMakeFiles/digidocpp_util.dir/util/File.cpp.o
[  8%] Building CXX object src/CMakeFiles/digidocpp_util.dir/util/log.cpp.o
[  9%] Linking CXX static library libdigidocpp_util.a
[  9%] Built target digidocpp_util
[ 10%] Generating xml/ts_119612v020101_sie_xsd.cxx, xml/ts_119612v020101_sie_xsd.hxx
[ 11%] Generating xml/conf.cxx, xml/conf.hxx
[ 13%] Generating xml/OpenDocument_manifest.cxx, xml/OpenDocument_manifest.hxx
[ 14%] Generating xml/xmldsig-core-schema.cxx, xml/xmldsig-core-schema.hxx
[ 15%] Generating xml/XAdES01903v132-201601.cxx, xml/XAdES01903v132-201601.hxx
[ 16%] Generating xml/XAdES01903v141-201601.cxx, xml/XAdES01903v141-201601.hxx
[ 17%] Generating xml/en_31916201v010101.cxx, xml/en_31916201v010101.hxx
[ 19%] Generating xml/OpenDocument_dsig.cxx, xml/OpenDocument_dsig.hxx
[ 20%] Generating xml/xml.cxx, xml/xml.hxx
[ 21%] Generating xml/ts_119612v020201_201601xsd.cxx, xml/ts_119612v020201_201601xsd.hxx
[ 22%] Generating xml/ts_119612v020101_additionaltypes_xsd.cxx, xml/ts_119612v020101_additionaltypes_xsd.hxx
Scanning dependencies of target digidocpp_priv
[ 23%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/conf.cxx.o
[ 25%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/OpenDocument_manifest.cxx.o
[ 26%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/xmldsig-core-schema.cxx.o
[ 27%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/XAdES01903v132-201601.cxx.o
[ 28%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/XAdES01903v141-201601.cxx.o
[ 29%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/en_31916201v010101.cxx.o
[ 30%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/OpenDocument_dsig.cxx.o
[ 32%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/xml.cxx.o
[ 33%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/ts_119612v020201_201601xsd.cxx.o
[ 34%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/ts_119612v020101_additionaltypes_xsd.cxx.o
[ 35%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/ts_119612v020101_sie_xsd.cxx.o
[ 36%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/crypto/Connect.cpp.o
[ 38%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/crypto/Digest.cpp.o
[ 39%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/crypto/TSL.cpp.o
[ 40%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/crypto/X509Crypto.cpp.o
[ 41%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/util/DateTime.cpp.o
[ 42%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/AdditionalInformationType.cpp.o
[ 44%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/AnyType.cpp.o
[ 45%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/ExtensionType.cpp.o
[ 46%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/ObjectType.cpp.o
[ 47%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/SecureDOMParser.cpp.o
[ 48%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/UnsignedSignaturePropertiesType.cpp.o
[ 50%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/URIResolver.cpp.o
[ 51%] Linking CXX static library libdigidocpp_priv.a
[ 51%] Built target digidocpp_priv
[ 52%] Generating tslcert1.h from /home/avatar/synology/build/libdigidocpp/etc/tl-mp1.crt
xxd: tslcert1.crt: No such file or directory
make[2]: *** [src/CMakeFiles/digidocpp.dir/build.make:80: src/tslcert1.h] Error 2
make[1]: *** [CMakeFiles/Makefile2:272: src/CMakeFiles/digidocpp.dir/all] Error 2
make: *** [Makefile:180: all] Error 2
```

There you go: 
```
$ make VERBOSE=1;
/usr/bin/cmake -S/home/avatar/synology/build/libdigidocpp -B/home/avatar/synology/build/libdigidocpp --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /home/avatar/synology/build/libdigidocpp/CMakeFiles /home/avatar/synology/build/libdigidocpp/CMakeFiles/progress.marks
make -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/home/avatar/synology/build/libdigidocpp'
make -f CMakeFiles/docs.dir/build.make CMakeFiles/docs.dir/depend
make[2]: Entering directory '/home/avatar/synology/build/libdigidocpp'
cd /home/avatar/synology/build/libdigidocpp && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/avatar/synology/build/libdigidocpp /home/avatar/synology/build/libdigidocpp /home/avatar/synology/build/libdigidocpp /home/avatar/synology/build/libdigidocpp /home/avatar/synology/build/libdigidocpp/CMakeFiles/docs.dir/DependInfo.cmake --color=
Dependee "/home/avatar/synology/build/libdigidocpp/CMakeFiles/docs.dir/DependInfo.cmake" is newer than depender "/home/avatar/synology/build/libdigidocpp/CMakeFiles/docs.dir/depend.internal".
Dependee "/home/avatar/synology/build/libdigidocpp/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/home/avatar/synology/build/libdigidocpp/CMakeFiles/docs.dir/depend.internal".
Scanning dependencies of target docs
make[2]: Leaving directory '/home/avatar/synology/build/libdigidocpp'
make -f CMakeFiles/docs.dir/build.make CMakeFiles/docs.dir/build
make[2]: Entering directory '/home/avatar/synology/build/libdigidocpp'
[  1%] Generating API documentation with Doxygen
/usr/bin/doxygen /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `JAVADOC_BANNER         =' at line 200, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `PYTHON_DOCSTRING       =' at line 228, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `OPTIMIZE_OUTPUT_SLICE  =' at line 300, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `NUM_PROC_THREADS       =' at line 466, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `EXTRACT_PRIV_VIRTUAL   =' at line 492, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `RESOLVE_UNNAMED_PARAMS =' at line 536, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `SHOW_HEADERFILE        =' at line 609, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `WARN_IF_INCOMPLETE_DOC =' at line 827, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `DOCSET_FEEDURL         =' at line 1360, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `FULL_SIDEBAR           =' at line 1571, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `OBFUSCATE_EMAILS       =' at line 1602, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `HTML_FORMULA_FORMAT    =' at line 1613, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `FORMULA_MACROFILE      =' at line 1639, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `MATHJAX_VERSION        =' at line 1661, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `LATEX_MAKEINDEX_CMD    =' at line 1850, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `LATEX_EMOJI_DIRECTORY  =' at line 1985, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `XML_NS_MEMB_FILE_SCOPE =' at line 2119, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `DOT_UML_DETAILS        =' at line 2432, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `DOT_WRAP_THRESHOLD     =' at line 2441, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: ignoring unsupported tag `DIR_GRAPH_MAX_DEPTH    =' at line 2514, file /home/avatar/synology/build/libdigidocpp/Doxyfile
warning: argument `TEXT' for option CLASS_GRAPH is not a valid boolean value
Using the default: YES!
make[2]: Leaving directory '/home/avatar/synology/build/libdigidocpp'
[  1%] Built target docs
make -f src/CMakeFiles/minizip.dir/build.make src/CMakeFiles/minizip.dir/depend
make[2]: Entering directory '/home/avatar/synology/build/libdigidocpp'
cd /home/avatar/synology/build/libdigidocpp && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/avatar/synology/build/libdigidocpp /home/avatar/synology/build/libdigidocpp/src /home/avatar/synology/build/libdigidocpp /home/avatar/synology/build/libdigidocpp/src /home/avatar/synology/build/libdigidocpp/src/CMakeFiles/minizip.dir/DependInfo.cmake --color=
Dependee "/home/avatar/synology/build/libdigidocpp/src/CMakeFiles/minizip.dir/DependInfo.cmake" is newer than depender "/home/avatar/synology/build/libdigidocpp/src/CMakeFiles/minizip.dir/depend.internal".
Dependee "/home/avatar/synology/build/libdigidocpp/src/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/home/avatar/synology/build/libdigidocpp/src/CMakeFiles/minizip.dir/depend.internal".
Scanning dependencies of target minizip
make[2]: Leaving directory '/home/avatar/synology/build/libdigidocpp'
make -f src/CMakeFiles/minizip.dir/build.make src/CMakeFiles/minizip.dir/build
make[2]: Entering directory '/home/avatar/synology/build/libdigidocpp'
[  2%] Building C object src/CMakeFiles/minizip.dir/minizip/zip.c.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cc -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DNOCRYPT -DNOUNCRYPT -DRELEASE_VER=9 -DVER_SUFFIX=\"\"  -fPIC -fvisibility=hidden   -Wno-unused-parameter -Wno-unused-value -o CMakeFiles/minizip.dir/minizip/zip.c.o   -c /home/avatar/synology/build/libdigidocpp/src/minizip/zip.c
[  3%] Building C object src/CMakeFiles/minizip.dir/minizip/unzip.c.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cc -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DNOCRYPT -DNOUNCRYPT -DRELEASE_VER=9 -DVER_SUFFIX=\"\"  -fPIC -fvisibility=hidden   -Wno-unused-parameter -Wno-unused-value -o CMakeFiles/minizip.dir/minizip/unzip.c.o   -c /home/avatar/synology/build/libdigidocpp/src/minizip/unzip.c
[  4%] Building C object src/CMakeFiles/minizip.dir/minizip/ioapi.c.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cc -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DNOCRYPT -DNOUNCRYPT -DRELEASE_VER=9 -DVER_SUFFIX=\"\"  -fPIC -fvisibility=hidden   -Wno-unused-parameter -Wno-unused-value -o CMakeFiles/minizip.dir/minizip/ioapi.c.o   -c /home/avatar/synology/build/libdigidocpp/src/minizip/ioapi.c
[  5%] Linking C static library libminizip.a
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -P CMakeFiles/minizip.dir/cmake_clean_target.cmake
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -E cmake_link_script CMakeFiles/minizip.dir/link.txt --verbose=1
/usr/bin/ar qc libminizip.a  CMakeFiles/minizip.dir/minizip/zip.c.o CMakeFiles/minizip.dir/minizip/unzip.c.o CMakeFiles/minizip.dir/minizip/ioapi.c.o
/usr/bin/ranlib libminizip.a
make[2]: Leaving directory '/home/avatar/synology/build/libdigidocpp'
[  5%] Built target minizip
make -f src/CMakeFiles/digidocpp_util.dir/build.make src/CMakeFiles/digidocpp_util.dir/depend
make[2]: Entering directory '/home/avatar/synology/build/libdigidocpp'
cd /home/avatar/synology/build/libdigidocpp && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/avatar/synology/build/libdigidocpp /home/avatar/synology/build/libdigidocpp/src /home/avatar/synology/build/libdigidocpp /home/avatar/synology/build/libdigidocpp/src /home/avatar/synology/build/libdigidocpp/src/CMakeFiles/digidocpp_util.dir/DependInfo.cmake --color=
Dependee "/home/avatar/synology/build/libdigidocpp/src/CMakeFiles/digidocpp_util.dir/DependInfo.cmake" is newer than depender "/home/avatar/synology/build/libdigidocpp/src/CMakeFiles/digidocpp_util.dir/depend.internal".
Dependee "/home/avatar/synology/build/libdigidocpp/src/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/home/avatar/synology/build/libdigidocpp/src/CMakeFiles/digidocpp_util.dir/depend.internal".
Scanning dependencies of target digidocpp_util
make[2]: Leaving directory '/home/avatar/synology/build/libdigidocpp'
make -f src/CMakeFiles/digidocpp_util.dir/build.make src/CMakeFiles/digidocpp_util.dir/build
make[2]: Entering directory '/home/avatar/synology/build/libdigidocpp'
[  7%] Building CXX object src/CMakeFiles/digidocpp_util.dir/util/File.cpp.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -DDIGIDOCPP_CONFIG_DIR=\"/usr/local/etc/digidocpp\"  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_util.dir/util/File.cpp.o -c /home/avatar/synology/build/libdigidocpp/src/util/File.cpp
[  8%] Building CXX object src/CMakeFiles/digidocpp_util.dir/util/log.cpp.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_util.dir/util/log.cpp.o -c /home/avatar/synology/build/libdigidocpp/src/util/log.cpp
[  9%] Linking CXX static library libdigidocpp_util.a
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -P CMakeFiles/digidocpp_util.dir/cmake_clean_target.cmake
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -E cmake_link_script CMakeFiles/digidocpp_util.dir/link.txt --verbose=1
/usr/bin/ar qc libdigidocpp_util.a  CMakeFiles/digidocpp_util.dir/util/File.cpp.o CMakeFiles/digidocpp_util.dir/util/log.cpp.o
/usr/bin/ranlib libdigidocpp_util.a
make[2]: Leaving directory '/home/avatar/synology/build/libdigidocpp'
[  9%] Built target digidocpp_util
make -f src/CMakeFiles/digidocpp_priv.dir/build.make src/CMakeFiles/digidocpp_priv.dir/depend
make[2]: Entering directory '/home/avatar/synology/build/libdigidocpp'
[ 10%] Generating xml/ts_119612v020101_sie_xsd.cxx, xml/ts_119612v020101_sie_xsd.hxx
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -E make_directory /home/avatar/synology/build/libdigidocpp/src/xml
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/xsdcxx cxx-tree --type-naming ucc --function-naming lcc --generate-serialization --suppress-assignment --std c++11 --output-dir /home/avatar/synology/build/libdigidocpp/src/xml --root-element-none --accessor-regex /assert/assert_/ --modifier-regex /assert/assert_/ --namespace-map http://uri.etsi.org/02231/v2#=digidoc::tsl --namespace-map http://uri.etsi.org/TrstSvc/SvcInfoExt/eSigDir-1999-93-EC-TrustedList/#=digidoc::tsl --namespace-map http://uri.etsi.org/01903/v1.3.2#=digidoc::xades /home/avatar/synology/build/libdigidocpp/etc/schema/ts_119612v020101_sie_xsd.xsd
[ 11%] Generating xml/conf.cxx, xml/conf.hxx
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -E make_directory /home/avatar/synology/build/libdigidocpp/src/xml
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/xsdcxx cxx-tree --type-naming ucc --function-naming lcc --generate-serialization --suppress-assignment --std c++11 --output-dir /home/avatar/synology/build/libdigidocpp/src/xml --root-element configuration /home/avatar/synology/build/libdigidocpp/etc/schema/conf.xsd
[ 13%] Generating xml/OpenDocument_manifest.cxx, xml/OpenDocument_manifest.hxx
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -E make_directory /home/avatar/synology/build/libdigidocpp/src/xml
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/xsdcxx cxx-tree --type-naming ucc --function-naming lcc --generate-serialization --suppress-assignment --std c++11 --output-dir /home/avatar/synology/build/libdigidocpp/src/xml --root-element manifest --namespace-map urn:oasis:names:tc:opendocument:xmlns:manifest:1.0=digidoc::manifest /home/avatar/synology/build/libdigidocpp/etc/schema/OpenDocument_manifest.xsd
[ 14%] Generating xml/xmldsig-core-schema.cxx, xml/xmldsig-core-schema.hxx
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -E make_directory /home/avatar/synology/build/libdigidocpp/src/xml
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/xsdcxx cxx-tree --type-naming ucc --function-naming lcc --generate-serialization --suppress-assignment --std c++11 --output-dir /home/avatar/synology/build/libdigidocpp/src/xml --root-element-none --namespace-map http://www.w3.org/2000/09/xmldsig#=digidoc::dsig --custom-type ObjectType=/ObjectTypeBase --hxx-epilogue "#include <xml/ObjectType.h>" /home/avatar/synology/build/libdigidocpp/etc/schema/xmldsig-core-schema.xsd
[ 15%] Generating xml/XAdES01903v132-201601.cxx, xml/XAdES01903v132-201601.hxx
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -E make_directory /home/avatar/synology/build/libdigidocpp/src/xml
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/xsdcxx cxx-tree --type-naming ucc --function-naming lcc --generate-serialization --suppress-assignment --std c++11 --output-dir /home/avatar/synology/build/libdigidocpp/src/xml --root-element-none --namespace-map http://www.w3.org/2000/09/xmldsig#=digidoc::dsig --namespace-map http://uri.etsi.org/01903/v1.3.2#=digidoc::xades --custom-type AnyType=/AnyTypeBase --custom-type UnsignedSignaturePropertiesType=/UnsignedSignaturePropertiesTypeBase --ordered-type UnsignedSignaturePropertiesType --hxx-epilogue "#include <xml/AnyType.h>" --hxx-epilogue "#include <xml/UnsignedSignaturePropertiesType.h>" /home/avatar/synology/build/libdigidocpp/etc/schema/XAdES01903v132-201601.xsd
[ 16%] Generating xml/XAdES01903v141-201601.cxx, xml/XAdES01903v141-201601.hxx
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -E make_directory /home/avatar/synology/build/libdigidocpp/src/xml
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/xsdcxx cxx-tree --type-naming ucc --function-naming lcc --generate-serialization --suppress-assignment --std c++11 --output-dir /home/avatar/synology/build/libdigidocpp/src/xml --root-element-none --namespace-map http://www.w3.org/2000/09/xmldsig#=digidoc::dsig --namespace-map http://uri.etsi.org/01903/v1.3.2#=digidoc::xades --namespace-map http://uri.etsi.org/01903/v1.4.1#=digidoc::xadesv141 /home/avatar/synology/build/libdigidocpp/etc/schema/XAdES01903v141-201601.xsd
[ 17%] Generating xml/en_31916201v010101.cxx, xml/en_31916201v010101.hxx
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -E make_directory /home/avatar/synology/build/libdigidocpp/src/xml
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/xsdcxx cxx-tree --type-naming ucc --function-naming lcc --generate-serialization --suppress-assignment --std c++11 --output-dir /home/avatar/synology/build/libdigidocpp/src/xml --root-element XAdESSignatures --namespace-map http://www.w3.org/2000/09/xmldsig#=digidoc::dsig --namespace-map http://uri.etsi.org/01903/v1.3.2#=digidoc::xades --namespace-map http://uri.etsi.org/02918/v1.2.1#=digidoc::asic /home/avatar/synology/build/libdigidocpp/etc/schema/en_31916201v010101.xsd
[ 19%] Generating xml/OpenDocument_dsig.cxx, xml/OpenDocument_dsig.hxx
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -E make_directory /home/avatar/synology/build/libdigidocpp/src/xml
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/xsdcxx cxx-tree --type-naming ucc --function-naming lcc --generate-serialization --suppress-assignment --std c++11 --output-dir /home/avatar/synology/build/libdigidocpp/src/xml --root-element document-signatures --namespace-map urn:oasis:names:tc:opendocument:xmlns:digitalsignature:1.0=digidoc::asic --namespace-map http://www.w3.org/2000/09/xmldsig#=digidoc::dsig /home/avatar/synology/build/libdigidocpp/etc/schema/OpenDocument_dsig.xsd
[ 20%] Generating xml/xml.cxx, xml/xml.hxx
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -E make_directory /home/avatar/synology/build/libdigidocpp/src/xml
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/xsdcxx cxx-tree --type-naming ucc --function-naming lcc --generate-serialization --suppress-assignment --std c++11 --output-dir /home/avatar/synology/build/libdigidocpp/src/xml --root-element-none /home/avatar/synology/build/libdigidocpp/etc/schema/xml.xsd
[ 21%] Generating xml/ts_119612v020201_201601xsd.cxx, xml/ts_119612v020201_201601xsd.hxx
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -E make_directory /home/avatar/synology/build/libdigidocpp/src/xml
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/xsdcxx cxx-tree --type-naming ucc --function-naming lcc --generate-serialization --suppress-assignment --std c++11 --output-dir /home/avatar/synology/build/libdigidocpp/src/xml --root-element TrustServiceStatusList --namespace-map http://uri.etsi.org/02231/v2#=digidoc::tsl --namespace-map http://www.w3.org/2000/09/xmldsig#=digidoc::dsig --custom-type AdditionalInformationType=/AdditionalInformationTypeBase --custom-type ExtensionType=/ExtensionTypeBase --hxx-epilogue "#include <xml/ExtensionType.h>" --hxx-epilogue "#include <xml/AdditionalInformationType.h>" /home/avatar/synology/build/libdigidocpp/etc/schema/ts_119612v020201_201601xsd.xsd
[ 22%] Generating xml/ts_119612v020101_additionaltypes_xsd.cxx, xml/ts_119612v020101_additionaltypes_xsd.hxx
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -E make_directory /home/avatar/synology/build/libdigidocpp/src/xml
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/xsdcxx cxx-tree --type-naming ucc --function-naming lcc --generate-serialization --suppress-assignment --std c++11 --output-dir /home/avatar/synology/build/libdigidocpp/src/xml --root-element-none --namespace-map http://uri.etsi.org/02231/v2#=digidoc::tsl --namespace-map http://uri.etsi.org/02231/v2/additionaltypes#=digidoc::tsl --namespace-map http://uri.etsi.org/01903/v1.3.2#=digidoc::xades /home/avatar/synology/build/libdigidocpp/etc/schema/ts_119612v020101_additionaltypes_xsd.xsd
cd /home/avatar/synology/build/libdigidocpp && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/avatar/synology/build/libdigidocpp /home/avatar/synology/build/libdigidocpp/src /home/avatar/synology/build/libdigidocpp /home/avatar/synology/build/libdigidocpp/src /home/avatar/synology/build/libdigidocpp/src/CMakeFiles/digidocpp_priv.dir/DependInfo.cmake --color=
Dependee "/home/avatar/synology/build/libdigidocpp/src/CMakeFiles/digidocpp_priv.dir/DependInfo.cmake" is newer than depender "/home/avatar/synology/build/libdigidocpp/src/CMakeFiles/digidocpp_priv.dir/depend.internal".
Dependee "/home/avatar/synology/build/libdigidocpp/src/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/home/avatar/synology/build/libdigidocpp/src/CMakeFiles/digidocpp_priv.dir/depend.internal".
Scanning dependencies of target digidocpp_priv
make[2]: Leaving directory '/home/avatar/synology/build/libdigidocpp'
make -f src/CMakeFiles/digidocpp_priv.dir/build.make src/CMakeFiles/digidocpp_priv.dir/build
make[2]: Entering directory '/home/avatar/synology/build/libdigidocpp'
[ 23%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/conf.cxx.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/conf.cxx.o -c /home/avatar/synology/build/libdigidocpp/src/xml/conf.cxx
[ 25%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/OpenDocument_manifest.cxx.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/OpenDocument_manifest.cxx.o -c /home/avatar/synology/build/libdigidocpp/src/xml/OpenDocument_manifest.cxx
[ 26%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/xmldsig-core-schema.cxx.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/xmldsig-core-schema.cxx.o -c /home/avatar/synology/build/libdigidocpp/src/xml/xmldsig-core-schema.cxx
[ 27%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/XAdES01903v132-201601.cxx.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/XAdES01903v132-201601.cxx.o -c /home/avatar/synology/build/libdigidocpp/src/xml/XAdES01903v132-201601.cxx
[ 28%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/XAdES01903v141-201601.cxx.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/XAdES01903v141-201601.cxx.o -c /home/avatar/synology/build/libdigidocpp/src/xml/XAdES01903v141-201601.cxx
[ 29%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/en_31916201v010101.cxx.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/en_31916201v010101.cxx.o -c /home/avatar/synology/build/libdigidocpp/src/xml/en_31916201v010101.cxx
[ 30%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/OpenDocument_dsig.cxx.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/OpenDocument_dsig.cxx.o -c /home/avatar/synology/build/libdigidocpp/src/xml/OpenDocument_dsig.cxx
[ 32%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/xml.cxx.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/xml.cxx.o -c /home/avatar/synology/build/libdigidocpp/src/xml/xml.cxx
[ 33%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/ts_119612v020201_201601xsd.cxx.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/ts_119612v020201_201601xsd.cxx.o -c /home/avatar/synology/build/libdigidocpp/src/xml/ts_119612v020201_201601xsd.cxx
[ 34%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/ts_119612v020101_additionaltypes_xsd.cxx.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/ts_119612v020101_additionaltypes_xsd.cxx.o -c /home/avatar/synology/build/libdigidocpp/src/xml/ts_119612v020101_additionaltypes_xsd.cxx
[ 35%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/ts_119612v020101_sie_xsd.cxx.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/ts_119612v020101_sie_xsd.cxx.o -c /home/avatar/synology/build/libdigidocpp/src/xml/ts_119612v020101_sie_xsd.cxx
[ 36%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/crypto/Connect.cpp.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/crypto/Connect.cpp.o -c /home/avatar/synology/build/libdigidocpp/src/crypto/Connect.cpp
[ 38%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/crypto/Digest.cpp.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/crypto/Digest.cpp.o -c /home/avatar/synology/build/libdigidocpp/src/crypto/Digest.cpp
[ 39%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/crypto/TSL.cpp.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/crypto/TSL.cpp.o -c /home/avatar/synology/build/libdigidocpp/src/crypto/TSL.cpp
[ 40%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/crypto/X509Crypto.cpp.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/crypto/X509Crypto.cpp.o -c /home/avatar/synology/build/libdigidocpp/src/crypto/X509Crypto.cpp
[ 41%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/util/DateTime.cpp.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/util/DateTime.cpp.o -c /home/avatar/synology/build/libdigidocpp/src/util/DateTime.cpp
[ 42%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/AdditionalInformationType.cpp.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/AdditionalInformationType.cpp.o -c /home/avatar/synology/build/libdigidocpp/src/xml/AdditionalInformationType.cpp
[ 44%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/AnyType.cpp.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/AnyType.cpp.o -c /home/avatar/synology/build/libdigidocpp/src/xml/AnyType.cpp
[ 45%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/ExtensionType.cpp.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/ExtensionType.cpp.o -c /home/avatar/synology/build/libdigidocpp/src/xml/ExtensionType.cpp
[ 46%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/ObjectType.cpp.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/ObjectType.cpp.o -c /home/avatar/synology/build/libdigidocpp/src/xml/ObjectType.cpp
[ 47%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/SecureDOMParser.cpp.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/SecureDOMParser.cpp.o -c /home/avatar/synology/build/libdigidocpp/src/xml/SecureDOMParser.cpp
[ 48%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/UnsignedSignaturePropertiesType.cpp.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/UnsignedSignaturePropertiesType.cpp.o -c /home/avatar/synology/build/libdigidocpp/src/xml/UnsignedSignaturePropertiesType.cpp
[ 50%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/xml/URIResolver.cpp.o
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/c++  -DBUILD_DATE=\"09.10.2022\" -DBUILD_VER=0 -DMAJOR_VER=3 -DMINOR_VER=14 -DRELEASE_VER=9 -DVER_SUFFIX=\"\" -Ddigidocpp_EXPORTS -I/home/avatar/synology/build/libdigidocpp/src  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -std=gnu++1z -o CMakeFiles/digidocpp_priv.dir/xml/URIResolver.cpp.o -c /home/avatar/synology/build/libdigidocpp/src/xml/URIResolver.cpp
[ 51%] Linking CXX static library libdigidocpp_priv.a
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -P CMakeFiles/digidocpp_priv.dir/cmake_clean_target.cmake
cd /home/avatar/synology/build/libdigidocpp/src && /usr/bin/cmake -E cmake_link_script CMakeFiles/digidocpp_priv.dir/link.txt --verbose=1
/usr/bin/ar qc libdigidocpp_priv.a  CMakeFiles/digidocpp_priv.dir/xml/conf.cxx.o CMakeFiles/digidocpp_priv.dir/xml/OpenDocument_manifest.cxx.o CMakeFiles/digidocpp_priv.dir/xml/xmldsig-core-schema.cxx.o CMakeFiles/digidocpp_priv.dir/xml/XAdES01903v132-201601.cxx.o CMakeFiles/digidocpp_priv.dir/xml/XAdES01903v141-201601.cxx.o CMakeFiles/digidocpp_priv.dir/xml/en_31916201v010101.cxx.o CMakeFiles/digidocpp_priv.dir/xml/OpenDocument_dsig.cxx.o CMakeFiles/digidocpp_priv.dir/xml/xml.cxx.o CMakeFiles/digidocpp_priv.dir/xml/ts_119612v020201_201601xsd.cxx.o CMakeFiles/digidocpp_priv.dir/xml/ts_119612v020101_additionaltypes_xsd.cxx.o CMakeFiles/digidocpp_priv.dir/xml/ts_119612v020101_sie_xsd.cxx.o CMakeFiles/digidocpp_priv.dir/crypto/Connect.cpp.o CMakeFiles/digidocpp_priv.dir/crypto/Digest.cpp.o CMakeFiles/digidocpp_priv.dir/crypto/TSL.cpp.o CMakeFiles/digidocpp_priv.dir/crypto/X509Crypto.cpp.o CMakeFiles/digidocpp_priv.dir/util/DateTime.cpp.o CMakeFiles/digidocpp_priv.dir/xml/AdditionalInformationType.cpp.o CMakeFiles/digidocpp_priv.dir/xml/AnyType.cpp.o CMakeFiles/digidocpp_priv.dir/xml/ExtensionType.cpp.o CMakeFiles/digidocpp_priv.dir/xml/ObjectType.cpp.o CMakeFiles/digidocpp_priv.dir/xml/SecureDOMParser.cpp.o CMakeFiles/digidocpp_priv.dir/xml/UnsignedSignaturePropertiesType.cpp.o CMakeFiles/digidocpp_priv.dir/xml/URIResolver.cpp.o
/usr/bin/ranlib libdigidocpp_priv.a
make[2]: Leaving directory '/home/avatar/synology/build/libdigidocpp'
[ 51%] Built target digidocpp_priv
make -f src/CMakeFiles/digidocpp.dir/build.make src/CMakeFiles/digidocpp.dir/depend
make[2]: Entering directory '/home/avatar/synology/build/libdigidocpp'
[ 52%] Generating tslcert1.h from /home/avatar/synology/build/libdigidocpp/etc/tl-mp1.crt
cd /home/avatar/synology/build/libdigidocpp/src && ln -sf /home/avatar/synology/build/libdigidocpp/etc/tl-mp1.crt tslcert1.crt
cd /home/avatar/synology/build/libdigidocpp/src && xxd -i tslcert1.crt tslcert1.h
xxd: tslcert1.crt: No such file or directory
make[2]: *** [src/CMakeFiles/digidocpp.dir/build.make:80: src/tslcert1.h] Error 2
make[2]: Leaving directory '/home/avatar/synology/build/libdigidocpp'
make[1]: *** [CMakeFiles/Makefile2:272: src/CMakeFiles/digidocpp.dir/all] Error 2
make[1]: Leaving directory '/home/avatar/synology/build/libdigidocpp'
make: *** [Makefile:180: all] Error 2
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -184,6 +184,10 @@ jobs:
         VCPKG_INSTALLED_DIR: ${{ github.workspace }}/vcpkg_installed
     - name: Install dependencies
       run: choco install doxygen.install swig -y > $null
+    - uses: actions/setup-python@v4
+      with:
+        python-version: 3.11
+        architecture: x86
     - name: Build xsd
       run: .\prepare_win_build_environment.ps1 -xsd
     - name: Build

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -1,1 +1,0 @@
-Please read the [common contributing guidelines](https://github.com/open-eid/org/blob/master/CONTRIBUTING.md) before you continue!

--- a/etc/Doxyfile.in
+++ b/etc/Doxyfile.in
@@ -1,4 +1,4 @@
-# Doxyfile 1.9.5
+# Doxyfile 1.9.7
 
 # This file describes the settings to be used by the documentation system
 # doxygen (www.doxygen.org) for a project.
@@ -86,7 +86,7 @@ CREATE_SUBDIRS         = NO
 # level increment doubles the number of directories, resulting in 4096
 # directories at level 8 which is the default and also the maximum value. The
 # sub-directories are organized in 2 levels, the first level always has a fixed
-# numer of 16 directories.
+# number of 16 directories.
 # Minimum value: 0, maximum value: 8, default value: 8.
 # This tag requires that the tag CREATE_SUBDIRS is set to YES.
 
@@ -363,6 +363,17 @@ MARKDOWN_SUPPORT       = YES
 
 TOC_INCLUDE_HEADINGS   = 5
 
+# The MARKDOWN_ID_STYLE tag can be used to specify the algorithm used to
+# generate identifiers for the Markdown headings. Note: Every identifier is
+# unique.
+# Possible values are: DOXYGEN Use a fixed 'autotoc_md' string followed by a
+# sequence number starting at 0. and GITHUB Use the lower case version of title
+# with any whitespace replaced by '-' and punctations characters removed..
+# The default value is: DOXYGEN.
+# This tag requires that the tag MARKDOWN_SUPPORT is set to YES.
+
+MARKDOWN_ID_STYLE      = DOXYGEN
+
 # When enabled doxygen tries to link words that correspond to documented
 # classes, or namespaces to their corresponding documentation. Such a link can
 # be prevented in individual cases by putting a % sign in front of the word or
@@ -487,6 +498,14 @@ LOOKUP_CACHE_SIZE      = 0
 
 NUM_PROC_THREADS       = 1
 
+# If the TIMESTAMP tag is set different from NO then each generated page will
+# contain the date or date and time when the page was generated. Setting this to
+# NO can help when comparing the output of multiple runs.
+# Possible values are: YES, NO, DATETIME and DATE.
+# The default value is: NO.
+
+TIMESTAMP              = YES
+
 #---------------------------------------------------------------------------
 # Build related configuration options
 #---------------------------------------------------------------------------
@@ -568,7 +587,8 @@ HIDE_UNDOC_MEMBERS     = NO
 # If the HIDE_UNDOC_CLASSES tag is set to YES, doxygen will hide all
 # undocumented classes that are normally visible in the class hierarchy. If set
 # to NO, these classes will be included in the various overviews. This option
-# has no effect if EXTRACT_ALL is enabled.
+# will also hide undocumented C++ concepts if enabled. This option has no effect
+# if EXTRACT_ALL is enabled.
 # The default value is: NO.
 
 HIDE_UNDOC_CLASSES     = NO
@@ -859,11 +879,26 @@ WARN_IF_INCOMPLETE_DOC = YES
 
 WARN_NO_PARAMDOC       = NO
 
+# If WARN_IF_UNDOC_ENUM_VAL option is set to YES, doxygen will warn about
+# undocumented enumeration values. If set to NO, doxygen will accept
+# undocumented enumeration values. If EXTRACT_ALL is set to YES then this flag
+# will automatically be disabled.
+# The default value is: NO.
+
+WARN_IF_UNDOC_ENUM_VAL = NO
+
 # If the WARN_AS_ERROR tag is set to YES then doxygen will immediately stop when
 # a warning is encountered. If the WARN_AS_ERROR tag is set to FAIL_ON_WARNINGS
 # then doxygen will continue running as if WARN_AS_ERROR tag is set to NO, but
 # at the end of the doxygen process doxygen will return with a non-zero status.
-# Possible values are: NO, YES and FAIL_ON_WARNINGS.
+# If the WARN_AS_ERROR tag is set to FAIL_ON_WARNINGS_PRINT then doxygen behaves
+# like FAIL_ON_WARNINGS but in case no WARN_LOGFILE is defined doxygen will not
+# write the warning messages in between other messages but write them at the end
+# of a run, in case a WARN_LOGFILE is defined the warning messages will be
+# besides being in the defined file also be shown at the end of a run, unless
+# the WARN_LOGFILE is defined as - i.e. standard output (stdout) in that case
+# the behavior will remain as with the setting FAIL_ON_WARNINGS.
+# Possible values are: NO, YES, FAIL_ON_WARNINGS and FAIL_ON_WARNINGS_PRINT.
 # The default value is: NO.
 
 WARN_AS_ERROR          = NO
@@ -1014,9 +1049,6 @@ EXCLUDE_PATTERNS       = */crypto/Connect.* \
 # output. The symbol name can be a fully qualified name, a word, or if the
 # wildcard * is used, a substring. Examples: ANamespace, AClass,
 # ANamespace::AClass, ANamespace::*Test
-#
-# Note that the wildcards are matched against the file with absolute path, so to
-# exclude all test directories use the pattern */test/*
 
 EXCLUDE_SYMBOLS        = digidoc::*Private \
                          digidoc::asic \
@@ -1220,10 +1252,11 @@ VERBATIM_HEADERS       = NO
 
 ALPHABETICAL_INDEX     = YES
 
-# In case all classes in a project start with a common prefix, all classes will
-# be put under the same header in the alphabetical index. The IGNORE_PREFIX tag
-# can be used to specify a prefix (or a list of prefixes) that should be ignored
-# while generating the index headers.
+# The IGNORE_PREFIX tag can be used to specify a prefix (or a list of prefixes)
+# that should be ignored while generating the index headers. The IGNORE_PREFIX
+# tag works for classes, function and member names. The entity will be placed in
+# the alphabetical list under the first letter of the entity name that remains
+# after removing the prefix.
 # This tag requires that the tag ALPHABETICAL_INDEX is set to YES.
 
 IGNORE_PREFIX          =
@@ -1302,7 +1335,12 @@ HTML_STYLESHEET        =
 # Doxygen will copy the style sheet files to the output directory.
 # Note: The order of the extra style sheet files is of importance (e.g. the last
 # style sheet in the list overrules the setting of the previous ones in the
-# list). For an example see the documentation.
+# list).
+# Note: Since the styling of scrollbars can currently not be overruled in
+# Webkit/Chromium, the styling will be left out of the default doxygen.css if
+# one or more extra stylesheets have been specified. So if scrollbar
+# customization is desired it has to be added explicitly. For an example see the
+# documentation.
 # This tag requires that the tag GENERATE_HTML is set to YES.
 
 HTML_EXTRA_STYLESHEET  = @CMAKE_SOURCE_DIR@/etc/Doxyfile.css
@@ -1318,17 +1356,13 @@ HTML_EXTRA_STYLESHEET  = @CMAKE_SOURCE_DIR@/etc/Doxyfile.css
 HTML_EXTRA_FILES       =
 
 # The HTML_COLORSTYLE tag can be used to specify if the generated HTML output
-# should be rendered with a dark or light theme. Default setting AUTO_LIGHT
-# enables light output unless the user preference is dark output. Other options
-# are DARK to always use dark mode, LIGHT to always use light mode, AUTO_DARK to
-# default to dark mode unless the user prefers light mode, and TOGGLE to let the
-# user toggle between dark and light mode via a button.
-# Possible values are: LIGHT Always generate light output., DARK Always generate
-# dark output., AUTO_LIGHT Automatically set the mode according to the user
-# preference, use light mode if no preference is set (the default)., AUTO_DARK
-# Automatically set the mode according to the user preference, use dark mode if
-# no preference is set. and TOGGLE Allow to user to switch between light and
-# dark mode via a button..
+# should be rendered with a dark or light theme.
+# Possible values are: LIGHT always generate light mode output, DARK always
+# generate dark mode output, AUTO_LIGHT automatically set the mode according to
+# the user preference, use light mode if no preference is set (the default),
+# AUTO_DARK automatically set the mode according to the user preference, use
+# dark mode if no preference is set and TOGGLE allow to user to switch between
+# light and dark mode via a button.
 # The default value is: AUTO_LIGHT.
 # This tag requires that the tag GENERATE_HTML is set to YES.
 
@@ -1363,15 +1397,6 @@ HTML_COLORSTYLE_SAT    = 100
 # This tag requires that the tag GENERATE_HTML is set to YES.
 
 HTML_COLORSTYLE_GAMMA  = 80
-
-# If the HTML_TIMESTAMP tag is set to YES then the footer of each generated HTML
-# page will contain the date and time when the page was generated. Setting this
-# to YES can help to show when doxygen was last run and thus if the
-# documentation is up to date.
-# The default value is: NO.
-# This tag requires that the tag GENERATE_HTML is set to YES.
-
-HTML_TIMESTAMP         = YES
 
 # If the HTML_DYNAMIC_MENUS tag is set to YES then the generated HTML
 # documentation will contain a main index with vertical navigation menus that
@@ -1521,6 +1546,16 @@ BINARY_TOC             = NO
 # This tag requires that the tag GENERATE_HTMLHELP is set to YES.
 
 TOC_EXPAND             = NO
+
+# The SITEMAP_URL tag is used to specify the full URL of the place where the
+# generated documentation will be placed on the server by the user during the
+# deployment of the documentation. The generated sitemap is called sitemap.xml
+# and placed on the directory specified by HTML_OUTPUT. In case no SITEMAP_URL
+# is specified no sitemap is generated. For information about the sitemap
+# protocol see https://www.sitemaps.org
+# This tag requires that the tag GENERATE_HTML is set to YES.
+
+SITEMAP_URL            =
 
 # If the GENERATE_QHP tag is set to YES and both QHP_NAMESPACE and
 # QHP_VIRTUAL_FOLDER are set, an additional index file will be generated that
@@ -2010,9 +2045,16 @@ PDF_HYPERLINKS         = YES
 
 USE_PDFLATEX           = YES
 
-# If the LATEX_BATCHMODE tag is set to YES, doxygen will add the \batchmode
-# command to the generated LaTeX files. This will instruct LaTeX to keep running
-# if errors occur, instead of asking the user for help.
+# The LATEX_BATCHMODE tag ignals the behavior of LaTeX in case of an error.
+# Possible values are: NO same as ERROR_STOP, YES same as BATCH, BATCH In batch
+# mode nothing is printed on the terminal, errors are scrolled as if <return> is
+# hit at every error; missing files that TeX tries to input or request from
+# keyboard input (\read on a not open input stream) cause the job to abort,
+# NON_STOP In nonstop mode the diagnostic message will appear on the terminal,
+# but there is no possibility of user interaction just like in batch mode,
+# SCROLL In scroll mode, TeX will stop only for missing files to input or if
+# keyboard input is necessary and ERROR_STOP In errorstop mode, TeX will stop at
+# each error, asking for user intervention.
 # The default value is: NO.
 # This tag requires that the tag GENERATE_LATEX is set to YES.
 
@@ -2032,14 +2074,6 @@ LATEX_HIDE_INDICES     = NO
 # This tag requires that the tag GENERATE_LATEX is set to YES.
 
 LATEX_BIB_STYLE        = plain
-
-# If the LATEX_TIMESTAMP tag is set to YES then the footer of each generated
-# page will contain the date and time when the page was generated. Setting this
-# to NO can help when comparing the output of multiple runs.
-# The default value is: NO.
-# This tag requires that the tag GENERATE_LATEX is set to YES.
-
-LATEX_TIMESTAMP        = NO
 
 # The LATEX_EMOJI_DIRECTORY tag is used to specify the (relative or absolute)
 # path from which the emoji images will be read. If a relative path is entered,
@@ -2206,7 +2240,7 @@ DOCBOOK_OUTPUT         = docbook
 #---------------------------------------------------------------------------
 
 # If the GENERATE_AUTOGEN_DEF tag is set to YES, doxygen will generate an
-# AutoGen Definitions (see http://autogen.sourceforge.net/) file that captures
+# AutoGen Definitions (see https://autogen.sourceforge.net/) file that captures
 # the structure of the code including all documentation. Note that this feature
 # is still experimental and incomplete at the moment.
 # The default value is: NO.
@@ -2377,15 +2411,8 @@ EXTERNAL_GROUPS        = YES
 EXTERNAL_PAGES         = YES
 
 #---------------------------------------------------------------------------
-# Configuration options related to the dot tool
+# Configuration options related to diagram generator tools
 #---------------------------------------------------------------------------
-
-# You can include diagrams made with dia in doxygen documentation. Doxygen will
-# then run dia to produce the diagram and insert it in the documentation. The
-# DIA_PATH tag allows you to specify the directory where the dia binary resides.
-# If left empty dia is assumed to be found in the default search path.
-
-DIA_PATH               =
 
 # If set to YES the inheritance and collaboration graphs will hide inheritance
 # and usage relations if the target is undocumented or is not a class.
@@ -2395,7 +2422,7 @@ HIDE_UNDOC_RELATIONS   = YES
 
 # If you set the HAVE_DOT tag to YES then doxygen will assume the dot tool is
 # available from the path. This tool is part of Graphviz (see:
-# http://www.graphviz.org/), a graph visualization toolkit from AT&T and Lucent
+# https://www.graphviz.org/), a graph visualization toolkit from AT&T and Lucent
 # Bell Labs. The other options in this section have no effect if this option is
 # set to NO
 # The default value is: NO.
@@ -2448,13 +2475,15 @@ DOT_NODE_ATTR          = "shape=box,height=0.2,width=0.4"
 
 DOT_FONTPATH           =
 
-# If the CLASS_GRAPH tag is set to YES (or GRAPH) then doxygen will generate a
-# graph for each documented class showing the direct and indirect inheritance
-# relations. In case HAVE_DOT is set as well dot will be used to draw the graph,
-# otherwise the built-in generator will be used. If the CLASS_GRAPH tag is set
-# to TEXT the direct and indirect inheritance relations will be shown as texts /
-# links.
-# Possible values are: NO, YES, TEXT and GRAPH.
+# If the CLASS_GRAPH tag is set to YES or GRAPH or BUILTIN then doxygen will
+# generate a graph for each documented class showing the direct and indirect
+# inheritance relations. In case the CLASS_GRAPH tag is set to YES or GRAPH and
+# HAVE_DOT is enabled as well, then dot will be used to draw the graph. In case
+# the CLASS_GRAPH tag is set to YES and HAVE_DOT is disabled or if the
+# CLASS_GRAPH tag is set to BUILTIN, then the built-in generator will be used.
+# If the CLASS_GRAPH tag is set to TEXT the direct and indirect inheritance
+# relations will be shown as texts / links.
+# Possible values are: NO, YES, TEXT, GRAPH and BUILTIN.
 # The default value is: YES.
 
 CLASS_GRAPH            = TEXT
@@ -2595,7 +2624,7 @@ DIR_GRAPH_MAX_DEPTH    = 1
 # The DOT_IMAGE_FORMAT tag can be used to set the image format of the images
 # generated by dot. For an explanation of the image formats see the section
 # output formats in the documentation of the dot tool (Graphviz (see:
-# http://www.graphviz.org/)).
+# https://www.graphviz.org/)).
 # Note: If you choose svg you need to set HTML_FILE_EXTENSION to xhtml in order
 # to make the SVG files visible in IE 9+ (other browsers do not have this
 # requirement).
@@ -2632,11 +2661,12 @@ DOT_PATH               =
 
 DOTFILE_DIRS           =
 
-# The MSCFILE_DIRS tag can be used to specify one or more directories that
-# contain msc files that are included in the documentation (see the \mscfile
-# command).
+# You can include diagrams made with dia in doxygen documentation. Doxygen will
+# then run dia to produce the diagram and insert it in the documentation. The
+# DIA_PATH tag allows you to specify the directory where the dia binary resides.
+# If left empty dia is assumed to be found in the default search path.
 
-MSCFILE_DIRS           =
+DIA_PATH               =
 
 # The DIAFILE_DIRS tag can be used to specify one or more directories that
 # contain dia files that are included in the documentation (see the \diafile
@@ -2713,3 +2743,19 @@ GENERATE_LEGEND        = YES
 # The default value is: YES.
 
 DOT_CLEANUP            = YES
+
+# You can define message sequence charts within doxygen comments using the \msc
+# command. If the MSCGEN_TOOL tag is left empty (the default), then doxygen will
+# use a built-in version of mscgen tool to produce the charts. Alternatively,
+# the MSCGEN_TOOL tag can also specify the name an external tool. For instance,
+# specifying prog as the value, doxygen will call the tool as prog -T
+# <outfile_format> -o <outputfile> <inputfile>. The external tool should support
+# output file formats "png", "eps", "svg", and "ismap".
+
+MSCGEN_TOOL            =
+
+# The MSCFILE_DIRS tag can be used to specify one or more directories that
+# contain msc files that are included in the documentation (see the \mscfile
+# command).
+
+MSCFILE_DIRS           =

--- a/examples/android/app/src/main/AndroidManifest.xml
+++ b/examples/android/app/src/main/AndroidManifest.xml
@@ -2,8 +2,6 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android">
 
     <uses-permission android:name="android.permission.INTERNET" />
-    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
-    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
 
     <application
         android:allowBackup="true"

--- a/examples/android/app/src/main/java/ee/ria/libdigidocpp/MainActivity.java
+++ b/examples/android/app/src/main/java/ee/ria/libdigidocpp/MainActivity.java
@@ -26,6 +26,8 @@ import java.net.HttpURLConnection;
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 import java.security.cert.X509Certificate;
 import java.sql.Timestamp;
 import java.util.Objects;
@@ -45,16 +47,6 @@ public class MainActivity extends Activity {
 		System.loadLibrary("digidoc_java");
 	}
 
-	static private void saveToFile(InputStream in, String path) throws IOException {
-		try (FileOutputStream out = new FileOutputStream(path)) {
-			byte[] buffer = new byte[10240];
-			int count;
-			while ((count = in.read(buffer)) != -1) {
-				out.write(buffer, 0, count);
-			}
-		}
-	}
-
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
@@ -67,14 +59,14 @@ public class MainActivity extends Activity {
 			try (ZipInputStream zis = new ZipInputStream(getResources().openRawResource(R.raw.schema))) {
 				ZipEntry ze;
 				while ((ze = zis.getNextEntry()) != null) {
-					saveToFile(zis, cache + "/" + ze.getName());
+					Files.copy(zis, Paths.get(cache, ze.getName()));
 				}
 			}
 			try (InputStream in = getResources().openRawResource(R.raw.test)) {
-				saveToFile(in, cache + "/test.bdoc");
+				Files.copy(in, Paths.get(cache, "test.bdoc"));
 			}
 			try (ByteArrayInputStream bin = new ByteArrayInputStream(new byte[] {})) {
-				saveToFile(bin, cache + "/EE_T.xml");
+				Files.copy(bin, Paths.get(cache, "EE_T.xml"));
 			}
 		} catch (IOException e) {
 			e.printStackTrace();
@@ -224,7 +216,7 @@ public class MainActivity extends Activity {
 			}
 
 			try (FileInputStream in = new FileInputStream(cache + "/digidocpp.log")) {
-				saveToFile(in, Environment.getExternalStorageDirectory().getAbsolutePath() + "/digidocpp.log");
+				Files.copy(in, Paths.get(Environment.getExternalStorageDirectory().getAbsolutePath(), "digidocpp.log"));
 			}
 			try (FileOutputStream out = new FileOutputStream(Environment.getExternalStorageDirectory().getAbsolutePath() + "/result.json")) {
 				out.write(r.toString().getBytes(StandardCharsets.UTF_8));
@@ -235,11 +227,11 @@ public class MainActivity extends Activity {
 		findViewById(R.id.run).setEnabled(true);
 	}
 
-	private static void deleteRecursive(File fileOrDirectory) {
+	private static boolean deleteRecursive(File fileOrDirectory) {
 		if (fileOrDirectory.isDirectory())
 			for (File child : Objects.requireNonNull(fileOrDirectory.listFiles()))
 				deleteRecursive(child);
-		fileOrDirectory.delete();
+		return fileOrDirectory.delete();
 	}
 
 	static private abstract class URLTask extends AsyncTask<Void, Void, Exception> {
@@ -269,8 +261,8 @@ public class MainActivity extends Activity {
 		DownloadTask(MainActivity activity) {
 			super(activity);
 			path = new File(activity.getCacheDir() + "/validate");
-			deleteRecursive(path);
-			path.mkdir();
+			if (deleteRecursive(path))
+				path.mkdir();
 		}
 
 		@Override
@@ -285,7 +277,7 @@ public class MainActivity extends Activity {
 					ZipEntry ze;
 					while ((ze = zis.getNextEntry()) != null) {
 						try {
-							saveToFile(zis, path + "/" + ze.getName());
+							Files.copy(zis, Paths.get(path.getPath(), ze.getName()));
 						} catch (Exception e) {
 							e.printStackTrace();
 						}

--- a/examples/android/build.gradle
+++ b/examples/android/build.gradle
@@ -5,7 +5,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.4.2'
+        classpath 'com.android.tools.build:gradle:8.1.0'
     }
 }
 

--- a/examples/android/gradle.properties
+++ b/examples/android/gradle.properties
@@ -18,3 +18,6 @@ android.useAndroidX=true
 # Automatically convert third-party libraries to use AndroidX
 android.enableJetifier=false
 org.gradle.unsafe.configuration-cache=true
+android.defaults.buildfeatures.buildconfig=true
+android.nonTransitiveRClass=true
+android.nonFinalResIds=true

--- a/examples/android/gradle/wrapper/gradle-wrapper.properties
+++ b/examples/android/gradle/wrapper/gradle-wrapper.properties
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.0-all.zip

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -89,13 +89,13 @@ foreach(TSL_CERT ${TSL_CERTS})
         add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${BASE}.h
             COMMAND $<TARGET_FILE:embedfile> ${TSL_CERT} ${BASE}_crt ${BASE}.h
             COMMENT "Generating ${BASE}.h from ${TSL_CERT}"
-            )
+        )
     else()
         add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${BASE}.h
-            COMMAND ln -sf ${TSL_CERT} ${BASE}.crt
-            COMMAND xxd -i ${BASE}.crt ${BASE}.h
+            COMMAND xxd -i ${TSL_CERT} ${BASE}.h
+            COMMAND sed -ie 's!unsigned char .*[][]!unsigned char ${BASE}_crt[]!' ${BASE}.h
             COMMENT "Generating ${BASE}.h from ${TSL_CERT}"
-            )
+        )
     endif()
     file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/tslcerts.h "#include \"${BASE}.h\"\n")
 endforeach()
@@ -105,7 +105,7 @@ set(COUNT 0)
 foreach(TSL_CERT ${TSL_CERTS})
     math(EXPR COUNT "${COUNT}+1")
     set(BASE "tslcert${COUNT}")
-    file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/tslcerts.h "digidoc::X509Cert(${BASE}_crt, ${BASE}_crt_len, X509Cert::Pem),\n")
+    file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/tslcerts.h "digidoc::X509Cert(${BASE}_crt, std::size(${BASE}_crt), X509Cert::Pem),\n")
 endforeach()
 file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/tslcerts.h "};")
 

